runtime/registry.go

```diff
@@ -92,8 +95,11 @@ func (r *Runtime) DeleteInstance(ctx context.Context, instanceID string, dropDB
 
 	// Drop the underlying data store
 	if dropDB {
-		conn, err := r.connCache.get(ctx, instanceID, inst.OLAPDriver, inst.OLAPDSN)
+		c, shared, _ := r.OLAPDef(inst)
+		vars := r.variables(inst.OLAPDriver, c.Configs, inst.ResolveVariables())
+		conn, release, err := r.connCache.get(ctx, instanceID, c.Type, vars, shared)
```

Comment:
This should call into a generic function like `AcquireHandle` or something similar. We don't want to replicate the connection cache accessing logic too many places.

Maybe some more generic functions might be needed in there, like `TryAcquire` (only if open) or something like that.

Or a function like `r.ConnectorDef(inst.ID, inst.RepoDriver)` for resolving the connector definition without opening it